docs/source/_src_user_guide/state_propagation/propagation_setup/acceleration_models/thrust.rst

```diff
@@ -0,0 +1,301 @@
+
+   - :func:`~tudatpy.numerical_simulation.propagation_setup.acceleration.thrust_from_custom_function`: this method lets you specify a custom variable thrust vector as a function, a constant specific impulse, and the frame in which the thrust is defined (default: inertial orientation).
+   - :func:`~tudatpy.numerical_simulation.propagation_setup.acceleration.thrust_and_isp_from_custom_function`: this method adds the capability of specifying a custom variable specific impulse as a function.
+   - :func:`~tudatpy.numerical_simulation.propagation_setup.acceleration.thrust_from_direction_and_magnitude`: this method takes :class:`~tudatpy.numerical_simulation.propagation_setup.thrust.ThrustDirectionSettings` and :class:`~tudatpy.numerical_simulation.propagation_setup.thrust.ThrustMagnitudeSettings` as inputs. These thrust `direction are discussed further on this page: <#thrust-direction>`_ and `magnitude <#thrust-magnitude>`_. The settings are created using factory functions, in the same was as acceleration settings, environment settings, etc.
+   
+
+An typical representative example on how the thrust acceleration can be set up, using the available :func:`~tudatpy.numerical_simulation.propagation_setup.acceleration.thrust_from_direction_and_magnitude` function, is provided below:
+
+   .. tabs::
+
+      .. tab:: Python
+
+         The Tudat(Py) API docs give more details on the :class:`~tudatpy.numerical_simulation.propagation_setup.thrust.ThrustDirectionSettings` and :class:`~tudatpy.numerical_simulation.propagation_setup.thrust.ThrustMagnitudeSettings` classes.
+
+         .. literalinclude:: /_src_snippets/simulation/propagation_setup/thrust/thrust_accelerations_basic_syntax.py
+            :language: python
+
+      .. tab:: C++
+
+         .. literalinclude:: /_src_snippets/simulation/propagation_setup/thrust/thrust_accelerations_basic_syntax.cpp
+            :language: cp
+
+In the above code snippet, note that we define the thrust acceleration as one that the vehicle exerts on itself.
+
+Thrust direction
+~~~~~~~~~~~~~~~~
+
+Below, the different methods that have been implemented to define the direction of the thrust are outlined. Note that these settings are *only* relevant if you use the :func:`~tudatpy.numerical_simulation.propagation_setup.acceleration.thrust_from_direction_and_magnitude` function.
+
+In all of these methods, the thrust direction that is defined is always in the **inertial frame**, either directly or indirectly. It is important to realize that, when specifying a thrust direction, the vehicle orientation itself is automatically defined.
+The direction of the thrust in the body-fixed frame can be additionally defined when specifying the `thrust magnitude <#thrust-magnitude>`_ (note that this design is currently under review, and may well be refactored in the near future).
+
+
+Thrust direction from state guidance settings
```

Comment:
If the code snippets are removed, I'd consider changing all the ways to create the thrust direction (and also magnitude) simply to bold font instead of actual subsections. I'd do that, adding a link to the related entry in the API docs and a few lines of explanation, not more than that, to keep everything a bit more compact.